pkg/controller.v1beta1/experiment/experiment_controller.go

```diff
@@ -114,33 +114,19 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 // addWatch adds a new Controller to mgr with r as the reconcile.Reconciler
 func addWatch(mgr manager.Manager, c controller.Controller) error {
 	// Watch for changes to Experiment
-	err := c.Watch(&source.Kind{Type: &experimentsv1beta1.Experiment{}}, &handler.EnqueueRequestForObject{})
+	err := c.Watch(source.Kind(mgr.GetCache(), &experimentsv1beta1.Experiment{}), &handler.EnqueueRequestForObject{})
 	if err != nil {
 		log.Error(err, "Experiment watch failed")
 		return err
 	}
 
 	// Watch for trials for the experiments
-	err = c.Watch(
-		&source.Kind{Type: &trialsv1beta1.Trial{}},
-		&handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &experimentsv1beta1.Experiment{},
-		})
-
-	if err != nil {
+	eventHandler := handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &experimentsv1beta1.Experiment{}, handler.OnlyControllerOwner())
+	if err = c.Watch(source.Kind(mgr.GetCache(), &trialsv1beta1.Trial{}), eventHandler); err != nil {
```

Comment:
The handler.EnqueueRequestForOwner{} typed was replaced with handler.EnqueueRequestForOwner() function:

https://github.com/kubernetes-sigs/controller-runtime/blob/116a1b831fffe7ccc3c8145306c3e1a3b1b14ffa/pkg/handler/enqueue_owner.go#L50-L62